### PR TITLE
Add ouger as a git submodule

### PR DESCRIPTION
**Pull request by mresvanis**
This PR adds [ouger](https://github.com/rh-ecosystem-edge/ouger) as a git submodule and an additional Dockerfile stage for its building and copying into the recert release image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,11 +18,17 @@ RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
 RUN cargo build --release --bin recert
 
+FROM golang:1.19-bookworm as ouger-builder
+COPY . .
```
Comment (omertuc) [inline Dockerfile:22]: Avoid wildcard copies because they invalidate the cache on every tiny thing

File: Dockerfile [hunk 2]
```diff
@@ -18,11 +18,17 @@ RUN cargo chef cook --release --recipe-path recipe.json
...
+RUN git submodule update --init --recursive
```
Comment (omertuc) [inline Dockerfile:23]: I think it's best to put this responsibility on whoever is running the Dockerfile

**Review by omertuc — CHANGES_REQUESTED** [review]
Thank you, a few caching concerns

**Comment by omertuc** [comment]
Finally https://asciinema.org/a/i8KjIinSiaK8HlBoPmpcK1Cza

**Comment by mresvanis** [comment]
> Finally https://asciinema.org/a/i8KjIinSiaK8HlBoPmpcK1Cza

I'll open a PR with the updated `.gitignore` :+1:

**Comment by omertuc** [comment]
No need, there's already an ignore, I was just happy to delete it from the recert root :)

**Pull request merged**